### ISSUE TITLE
`cppcoreguidelines-missing-std-forward` should not warn on deleted functions

### ISSUE THREAD

**Issue by jonesmz** [llvm/llvm-project#106276]
E.g.

```cpp
class MyClass
{
    template<typename ... ARGS_T>
    void some_function(ARGS_T && ...) = delete; // warning: forwarding reference parameter '' is never forwarded inside the function body
};
```

The function has no body, there's no way to satisfy this warning.

**Comment by 5chmidti** [comment]
A fix for this has already been implemented and will be available with the LLVM 19 release
https://github.com/llvm/llvm-project/pull/83055
https://godbolt.org/z/WMoebj54n
